value-fixture/test/org/immutables/fixture/with/WithEnumsTest.java

```diff
@@ -0,0 +1,63 @@
+package org.immutables.fixture.with;
+
+import org.junit.Test;
+
+import java.math.RoundingMode;
+import java.util.Optional;
+
+import static org.immutables.check.Checkers.check;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class WithEnumsTest {
+  private static final ImmutableWithEnums TEST_IMMUTABLE = ImmutableWithEnums.builder()
+          .roundingMode(RoundingMode.CEILING)
+          .maybeRoundingMode(RoundingMode.HALF_DOWN)
+          .nullableRoundingMode(RoundingMode.UNNECESSARY)
+          .build();
+  private static final ImmutableWithEnums TEST_IMMUTABLE_WITH_NULLS = ImmutableWithEnums.builder()
+          .roundingMode(RoundingMode.DOWN)
+          .maybeRoundingMode(Optional.empty())
+          .nullableRoundingMode(null)
+          .build();
+
+  @Test
+  public void doNotAllowNullEnumValueInNonNullableWith() {
+    assertThrows(NullPointerException.class, () -> TEST_IMMUTABLE.withRoundingMode(null));
+  }
+
+  @Test
+  public void allowNullEnumValueInNullableWith() {
+    ImmutableWithEnums withNullableRoundingMode = TEST_IMMUTABLE.withNullableRoundingMode(null);
+    check(withNullableRoundingMode.getNullableRoundingMode()).isNull();
+  }
+
+  @Test
+  public void withOnEqualEnumValueIsSameInstance() {
+    check(TEST_IMMUTABLE.withRoundingMode(RoundingMode.CEILING)).same(TEST_IMMUTABLE);
+    check(TEST_IMMUTABLE.withMaybeRoundingMode(RoundingMode.HALF_DOWN)).same(TEST_IMMUTABLE);
+    check(TEST_IMMUTABLE.withNullableRoundingMode(RoundingMode.UNNECESSARY)).same(TEST_IMMUTABLE);
+  }
+
+  @Test
+  public void withOnNullEnumValueIsSameInstance() {
+    check(TEST_IMMUTABLE_WITH_NULLS.withNullableRoundingMode(null)).same(TEST_IMMUTABLE_WITH_NULLS);
+    check(TEST_IMMUTABLE_WITH_NULLS.withMaybeRoundingMode(Optional.empty())).same(TEST_IMMUTABLE_WITH_NULLS);
+  }
+
+  @Test
+  public void withOnDifferentEnumValueIsNotSameInstance() {
+    check(TEST_IMMUTABLE.withRoundingMode(RoundingMode.FLOOR)).not().same(TEST_IMMUTABLE);
```

Comment:
I would even check for `not().is()` (`!Objects.equals()`) which is stronger than `not().same()` (`a != b`)

Reply:
Shouldn't `same()` be adequate here since `with(...)` returns the same immutable instance if the newValue is equal? 
```java
public final ImmutableWithEnums withRoundingMode(RoundingMode value) {
    RoundingMode newValue = Objects.requireNonNull(value, "roundingMode");
    if (this.roundingMode == newValue) return this;
    return new ImmutableWithEnums(newValue, this.nullableRoundingMode, this.maybeRoundingMode);
  }
```
i.e. we're testing whether the above `if` statement falls through for `with` methods?

As far as value equality with `Objects.equals()`, the behavior should be identical to before as I didn't change any of the code generation around  `equalTo`  on enum fields (only `with()` methods because I noticed they generated duplicate object equals checks). I initially intended to, but I wasn't able to get anything passing the existing test suite after a lot of trouble.

Reply:
I was thinking about this case : 

```java
public final ImmutableWithEnums withRoundingMode(RoundingMode value) {
      // ....
    return new ImmutableWithEnums(this.roundingMode, // hypothetical error in codegen (should be newValue really)
       this.nullableRoundingMode, this.maybeRoundingMode);
 }
```

`not().same()` would not catch it but `not().is()` would.

Reply:
I'm fine either way

Reply:
I see what you mean now 🤔. I can add some tests for that case as well 👍